homeassistant/components/soma/__init__.py

```diff
@@ -0,0 +1,102 @@
+"""Support for Soma Smartshades."""
+import logging
+
+import voluptuous as vol
+from api.soma_api import SomaApi
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import HomeAssistantType
+
+from homeassistant.const import (
+    CONF_HOST, CONF_PORT)
+
+from .const import DOMAIN, HOST, API
+
+
+DEVICES = 'devices'
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_PORT): cv.string
```

Comment:
We don't seem to use the port option?

Reply:
I will add that to the middleware library tomorrow. I had it hardcoded and forgot.